scripts/tests/sanitycheck/test_testsuite_class.py

```diff
@@ -38,3 +39,44 @@ def test_testsuite_add_testcases(class_testsuite):
     testcase = class_testsuite.testcases.get(tests_rel_dir + 'test_a/test_a.check_1')
     assert testcase.name == tests_rel_dir + 'test_a/test_a.check_1'
     assert all(isinstance(n, TestCase) for n in class_testsuite.testcases.values())
+
+@pytest.mark.parametrize("board_root_dir", [("board_config_file_not_exist"), ("board_config")])
+def test_add_configurations(test_data, class_testsuite, board_root_dir):
+    """ Testing add_configurations function of TestSuite class in Sanitycheck
+    Test : Asserting on default platforms list
+    """
+    class_testsuite.board_roots = os.path.abspath(test_data + board_root_dir)
+    suite = TestSuite(class_testsuite.board_roots, class_testsuite.roots, class_testsuite.outdir)
+    if board_root_dir == "board_config":
+        suite.add_configurations()
+        assert sorted(suite.default_platforms) == sorted(['demo_board_1', 'demo_board_3'])
+    elif board_root_dir == "board_config_file_not_exist":
+        suite.add_configurations()
+        assert sorted(suite.default_platforms) != sorted(['demo_board_1'])
+
+def test_get_all_testcases(class_testsuite, all_testcases_dict):
+    """ Testing get_all_testcases function of TestSuite class in Sanitycheck """
+    class_testsuite.testcases = all_testcases_dict
+    expected_tests = ['test_b.check_1', 'test_b.check_2', 'test_c.check_1', 'test_c.check_2', 'test_a.check_1', 'test_a.check_2', 'sample_test.app']
+    assert len(class_testsuite.get_all_tests()) == 7
+    assert sorted(class_testsuite.get_all_tests()) == sorted(expected_tests)
+
+def test_get_toolchain(class_testsuite, monkeypatch, capsys):
+    """ Testing get_toolchain function of TestSuite class in Sanitycheck
+    Test : Monkeypatch to delete the  ZEPHYR_TOOLCHAIN_VARIANT env var
+    and check if appropriate error is raised"""
+    toolchain = class_testsuite.get_toolchain()
+    assert toolchain in ["zephyr", "gnuarmemb"]
```

Comment:
here you are assuming that ZEPHYR_TOOLCHAIN_VARIANT is already set in the environment and is zephyr or gnuarmemb, this might fail if someone  has something else set or if the variable is not set.

Reply:
Removed the assumption that ZEPHYR_TOOLCHAIN_VARIANT can only be zephyr or gnuarmemb. This is now only testing for the error message if ZEPHYR_TOOLCHAIN_VARIANT is not in the environment. But testcase wont fail if it is not in env while running the testcases.

Reply:
the solution is not to delete assert completely, you could still set the environment there and keep the test, just do not rely on user environment.

Reply:
So get_toolchain() function reads the ZEPHYR_TOOLCHAIN_VARIANT from user environment and returns that toolchain. If I keep that test scenario then it will have to rely on user environment. Am i missing any info here ?

Reply:
you can mock the env. variable

Reply:
Fixed this and added the test scenario back by mocking the env. Thanks.